apps/tutorial-servo-can-control/src/TutorialServo.cpp

```diff
@@ -0,0 +1,51 @@
+#include "TutorialServo.h"
+#include "mbed.h"
+
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees, float minPulsewidthInMs ,
+              float maxPulsewidthInMs)
+                  : m_servoPwmOut(servoPin), m_servoRangeInDegrees(servoRangeInDegrees),
+                    m_minPulsewidthInMs(minPulsewidthInMs),
+                    m_maxPulsewidthInMs(maxPulsewidthInMs) {}
+
+void TutorialServo::setPositionInDegrees(const float degrees) {
+  //Ensure degree entered is within the range specified in m_servoRangeInDegrees
+   float fixValue(float value, float minValue, float maxValue){
+    if(value< minValue){
+      return minValue;
+    }
+    else if(value> maxValue){
+      return maxValue;
+    }
+    else{
+      return value;}
+  }

```

Comment:
This should give a compile error, you can't have a function inside a function. For helper functions, you can just define them as static private functions

Reply:
An alternative option would be to just use `std::clamp`, which does the same thing as what your `fixValue` function is doing: https://en.cppreference.com/w/cpp/algorithm/clamp

Reply:
Thanks, I'll check that out. We spoke about this on Thursday, and I changed it, but I haven't committed the change yet

Reply:
I have fixValue() function defined outside the function now but do I need to make it a static private function as well? I know you only stressed the importance of defining the fixValue() function outside the setPositionInDegrees() function on Thursday so I'm a little confused

Reply:
You can make it a private static function within the TutorialServo class